### ISSUE TITLE
Use a ThreadLocal<Codec> variable to contain charset codecs

### ISSUE THREAD

**Issue by jrte** [jrte/ribose#36]
Charset codecs are used throughout ribose code. Storing them in a `ThreadLocal<Codec>` on first use allows them to be accessed anywhere. `IModel.close()` implementations detach the `ThreadLocal<Codec>` instance from the calling thread. But 
threads that do not directly or indirectly (try-with-resources) call `IModel.close()`call the static `IModel.detach()` method to rid themselves of the thread local `Codec` instance.

**Comment by jrte** [comment]
Fix is implemented in commit f83daba. There are interface changes:

- IEffector
  - javadoc comment changes only
- IModel
  - add detach() method to clear ThreadLocal<Codec> from calling thread
- IOutput
  - removed decoder() and encoder() methods (use Codec methods)
- IParameterizedEffector
  - remove decoder parameter from showParamterTokens()
- ITarget
  - formatting changes only
- IToken
  - getName(CharsetDecoder) replaced with asString()
- ITransducrtor
  - formatting changes only
- BaseEffector
  - remove decoder() and encoder() methods (use Codec methods)
  - remove decoder parameter from showParameterTokens()
- BaseParameterizedEffector
  - remove decoder parameter from showParameterTokens()
- Bytes
  - add `public static final Bytes EMPTY = new Bytes(Bytes.EMPTY_BYTES)`
  - remove decode() and encode() methods (use Codec methods)
  - replace toString(CharsetDecoder) with asString()

**Comment by jrte** [comment]
Fixed in 87f6054.
